apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -92,139 +108,108 @@ public static System.Type GetDefaultClrTypeFromSpannerType(this TypeCode code)
             switch (code)
             {
                 case TypeCode.Bool:
-                    return typeof(bool);
+                    return typeof(bool?);
                 case TypeCode.Int64:
-                    return typeof(long);
+                    return typeof(long?);
                 case TypeCode.Float64:
-                    return typeof(double);
+                    return typeof(double?);
                 case TypeCode.Timestamp:
                 case TypeCode.Date:
-                    return typeof(DateTime);
+                    return typeof(DateTime?);
                 case TypeCode.String:
                     return typeof(string);
                 case TypeCode.Bytes:
                     return typeof(byte[]);
                 case TypeCode.Array:
-                    return typeof(IList);
+                    return typeof(ArrayList);
```

Comment:
Using `List<object>` [is recommended](https://msdn.microsoft.com/en-us/library/system.collections.arraylist(v=vs.110).aspx#Remarks); if there's no specific reason to use the non-generic `ArrayList`.